src/lib/apollo/accountListRedirect.test.ts

```diff
@@ -0,0 +1,67 @@
+import { GraphQLError } from 'graphql';
+import {
+  isAccountListNotFoundError,
+  replaceUrlAccountList,
+} from './accountListRedirect';
+
+describe('isAccountListNotFoundError', () => {
+  const makeError = (message: string, code: string): GraphQLError =>
+    new GraphQLError(message, { extensions: { code } });
+
+  it('returns false for errors besides NOT_FOUND', () => {
+    expect(
+      isAccountListNotFoundError(
+        makeError('Internal Server Error', 'INTERNAL_SERVER_ERROR'),
+      ),
+    ).toBe(false);
+  });
+
+  it('returns false for not found errors besides account lists', () => {
+    expect(
+      isAccountListNotFoundError(makeError('Contact not found', 'NOT_FOUND')),
+    ).toBe(false);
+  });
+
+  it('returns true for not found errors from the Rails API', () => {
+    expect(
+      isAccountListNotFoundError(
+        makeError('AccountList not found', 'NOT_FOUND'),
+      ),
+    ).toBe(true);
+  });
+
+  it('returns true for not found errors from the proxy API', () => {
+    expect(
+      isAccountListNotFoundError(
+        makeError("Resource 'AccountList' not found", 'NOT_FOUND'),
+      ),
+    ).toBe(true);
+  });
+});
+
+describe('replaceUrlAccountList', () => {
+  it('returns to default url if url is invalid', () => {
+    expect(replaceUrlAccountList('/', '12345')).toBe('/accountLists');
+    expect(replaceUrlAccountList('https://google.com', '12345')).toBe(
+      '/accountLists',
+    );
+  });
+
+  it('returns to default url if defaultAccountListId is invalid', () => {
+    expect(replaceUrlAccountList('/accountLists/', null)).toBe('/accountLists');
+  });
+
+  it('replaces the default accountListId in the url', () => {
+    expect(
+      replaceUrlAccountList('/accountList/invalid-account-list/', '12345'),
```

Comment:
Is this supposed to be `/accountLists/...` (pluralized)? If it's supposed to be an invalid pathname, let's use a url like `/invalid-url` and move it to `returns to default url if defaultAccountListId is invalid`.

Either way, it's probably good to have a test for `/accountLists/invalid-account-list/` too.